### ISSUE TITLE
Link error of CoMiSo and libigl in visual studio 2012

### ISSUE THREAD

**Issue by changjianli** [libigl/libigl#190]
Hi,

I use libel in visual studio 2012 with function igloo::comes::nosy(....), but there is link error: 
error LNK2019: unresolved external symbol COMISO::ConstrainedSolver::solve

The comiso library is compiled in visual studio 2012. How can I solve it?

Thanks

**Comment by danielepanozzo** [comment]
Hi,

It is difficult to debug linking errors with visual studio, could it be that you are mixing 32 and 64bit binaries?

We provide a version of Comiso with a custom cmake file that makes sure that all settings are correct for libigl in this repository: https://github.com/libigl/CoMISo

I tested it with VS2015 and it works out of the box, see the instructions here on how to use it with a clean libigl installation: http://libigl.github.io/libigl/tutorial/tutorial.html#chapter1:introductiontolibigl

**Comment by changjianli** [comment]
Hi Daniel,

Thanks a lot. I will try


Best,
Changjian

**Comment by changjianli** [comment]
Hi Daniel,

As you said, I still has the link error. My mac version works well, but I want to use it in VS. The lib version is 32bit, does it need 64bit?

Another problem is that how about blas version?

**Comment by danielepanozzo** [comment]
You have to consistently compile everything (libigl,comiso,blas) consistently (all 64 or all 32 bits). Everything also has to be compiled with the same version of visual studio, you cannot mix vs2012 and vs2015 binaries. 

I am surprised that it does not work out of the box with VS2015. Could you try to compile the libigl tutorial with VS2015 with a clean install of libigl and libigl/comiso?

**Comment by changjianli** [comment]
Actually, I did not compile libigl tutorial, I just include it in my project. And I compile everything in Visual Studio 2012(win 32). 

The problem is that when I use function nosy(..), it needs comiso solver. So I downloaded and compiled it as said here(https://www.graphics.rwth-aachen.de/software/comiso), but it has link error for the function constrainedsolver::solve(...). 

Now I changed to the one you recommended, used the simple cmake file, it compiles well, but still has link error.
